stdlib/builtins.pyi

```diff
@@ -1018,7 +1018,10 @@ class filter(Iterator[_T], Generic[_T]):
     def __next__(self) -> _T: ...
 
 def format(__value: object, __format_spec: str = ...) -> str: ...  # TODO unicode
-def getattr(__o: Any, name: str, __default: Any = ...) -> Any: ...
+@overload
+def getattr(__o: Any, name: str) -> Any: ...
+@overload
+def getattr(__o: Any, name: str, __default: _T = ...) -> Union[Any, _T]: ...
```

Comment:
Nit: You should explicitly say that the second case is not used for `getattr(foo, "bar")`

```suggestion
def getattr(__o: Any, name: str, __default: _T) -> Union[Any, _T]: ...
```

That said, your code already works, because `getattr(foo, "bar")` will choose the first matching overload, and won't look at the second overload.

Reply:
Marking as unresolved because Python 2 stubs need this too.

Reply:
I actually didn't know that that was how that worked, thanks for the commit! do we need to do that for the py2 stubs too?

Reply:
oh nice, your second comment didn't display for whatever reason.

Reply:
I don't see documentation for this ` = ...` syntax in either https://mypy.readthedocs.io/en/stable/kinds_of_types.html#callable-types or https://mypy.readthedocs.io/en/stable/stubs.html, by the way. It's new to me, too.

Reply:
This is the usual syntax for default arguments, as in `def foo(bar="lol")`. In stubs, the convention is to use `...` instead of the actual default value, and to add a type annotation for the argument, so it becomes something like `def foo(bar: str = ...)`. But that's just a convention, and `def foo(bar: str = "lol")` in a stub file would work too.

Reply:
ooooohhhhhh I just wasn't making the connection that the ellipsis would represent the individual default value for the individual kwarg; I just keep thinking that ellipses represent omitting big chunks of code or elements of iterables, or whatever.